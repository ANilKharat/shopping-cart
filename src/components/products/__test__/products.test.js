import React from 'react';
import { mount } from 'enzyme'
import ProductPage from '../Products'
import { Provider } from 'react-redux';
import store from '../../../redux/store/store'

describe('Products page test case', () => {
    it("Snapshot test case", () => {
        const component = mount(
            <Provider store={store}>
                <ProductPage />
            </Provider>
        )
        expect(component).toMatchSnapshot()
    })
    it("Button click test case", () => {
        const setProductToStore = jest.fn()
        const component = mount(
            <Provider store={store}>
                <ProductPage onClick={setProductToStore} />
            </Provider>
        )
        expect(component.find('button').at(0).simulate('click'))
    })
})
