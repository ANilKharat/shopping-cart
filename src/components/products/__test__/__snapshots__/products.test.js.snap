// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Products page test case Button click test case 1`] = `ReactWrapper {}`;

exports[`Products page test case Snapshot test case 1`] = `ReactWrapper {}`;
